web/frontend/src/components/slam-2d-render.vue

```diff
@@ -110,6 +155,15 @@ const updateCloud = (pointcloud: Uint8Array) => {
   intersectionPlane.position.set(center.x, 0, center.z);
   raycaster.objects = [intersectionPlane];
 
+  const colors = points.geometry.attributes.color;
+  // if the PCD has a color attribute defined, convert those colors using the colorMap
+  if (colors instanceof THREE.BufferAttribute || colors instanceof THREE.InterleavedBufferAttribute) {
+    for (let i = 0; i < colors.count; i += 1) {
+      const colorMapPoint = colorBuckets(colors.getZ(i));
+      colors.setXYZ(i, colorMapPoint.x, colorMapPoint.y, colorMapPoint.z);
```

Comment:
@micheal-parks Just confirming: Is this the idiomatic way to set the color scheme of points in three.js?

Reply:
Yep!

Reply:
@micheal-parks do you know why three.js uses setXYZ instead of something like setRGB for colors? Is it because the color type is a more generic `THREE.BufferAttribute`

Reply:
Exactly, it's because the `BufferAttribute` class is used for any all sorts of vector data, like colors, vertices, normals ,etc, and the authors haven't added a `setRGB` alias function